uniffi_bindgen/src/bindings/kotlin/templates/NamespaceLibraryTemplate.kt

```diff
@@ -29,12 +29,35 @@ internal interface _UniFFILib : Library {
             }
             {% endif %}
         }
+
+        internal val FUTURE_WAKER_ENVIRONMENTS: ConcurrentHashMap<Int, RustFutureWakerEnvironment<out Any>> by lazy {
+            ConcurrentHashMap(8)
+        }
     }
 
     {% for func in ci.iter_ffi_function_definitions() -%}
+    {%- if func.is_async() %}
```

Comment:
I'm a bit confused by some of this, but this is a different template context (note the 2 `#[derive(Template)]`s in kotlin/mod.rs). - NamespaceLibraryTemplate is `{% include`d in wrapper.kt - after wrapper.kt has rendered the imports! I'm only just starting to look at this, but I wonder if it can be done in `ObjectTemplate` when we know we have something async on our hands (I see Method has a new `is_async` - maybe Object could grow a new, say, "has_async_method" boolean so you can add the imports once in that file?

Reply:
The problem is that `ObjectTemplate` only handles methods, not functions. I think it needs a little bit more work, that I would like to postpone into a new follow-up PR.